pkg/codegen/schema/schema_test.go

```diff
@@ -996,3 +996,280 @@ func TestBindDefaultInt(t *testing.T) {
 		assert.Fail(t, "did not catch underflow")
 	}
 }
+
+func TestFunctionSpecToJSONAndYAMLTurnaround(t *testing.T) {
```

Comment:
Allright this turns out to be particularly tricky since json and yaml packages behave differently around the edges; hopefully this test matrix covers all the interesting variations.